services/frontend-service/src/ui/utils/store.tsx

```diff
@@ -553,6 +554,47 @@ export const useFilteredApplicationLocks = (appNameParam: string | null): Displa
     return sortLocks(filteredLocks, 'newestToOldest');
 };
 
+export interface DisplayApplicationLock {
+    lock: DisplayLock;
+    application: Environment_Application;
+    environment: Environment;
+    environmentGroup: EnvironmentGroup;
+}
+
+export const useDisplayApplicationLocks = (appName: string | null): DisplayApplicationLock[] => {
```

Comment:
We do something similar here at the component level, which allows the UI to be updated when a new lock is inserted: 
https://github.com/freiheit-com/kuberpult/blob/93a322c816e976de388316bcdf93afebe6e7b2e8/services/frontend-service/src/ui/Pages/Locks/LocksPage.tsx#L63

Please use the same approach